### PR TITLE
Add detect encoding with BOM: UTF-7 and GB-18030

### LINKED ISSUE #79: Add detect encoding with BOM: UTF-7 and GB-18030
Hello!

Add detect encoding with BOM: UTF-7 and GB-18030.

It seems that this can be done if you believe the table from the [wiki](https://en.wikipedia.org/wiki/Byte_order_mark#Byte_order_marks_by_encoding)

**UTF-7**:
- `2B 2F 76 38` (`43 47 118 56`)
- `2B 2F 76 39` (`43 47 118 57`)
- `2B 2F 76 2B` (`43 47 118 43`)
- `2B 2F 76 2F` (`43 47 118 47`)
- `2B 2F 76 38 2D` (`43 47 118 56 45`)

**GB-18030**:
- `84 31 95 33` (`132 49 149 51`)

### PR DESCRIPTION
**Pull request by rstm-sf**
Resolve #79

Add detect
- [x] UTF-7 BOM
- [x] GB-18030 BOM

### CODE REVIEW

**Comment by rstm-sf** [comment]
>  Simplification of checks in FindCharSetByBom
because len <= buf.Length always

So in the end it is called from the following places
https://github.com/CharsetDetector/UTF-unknown/blob/cb3dca2a51d4ccf769687cd18eebd12d60e8874b/src/CharsetDetector.cs#L142

https://github.com/CharsetDetector/UTF-unknown/blob/cb3dca2a51d4ccf769687cd18eebd12d60e8874b/src/CharsetDetector.cs#L199-L201

via

https://github.com/CharsetDetector/UTF-unknown/blob/cb3dca2a51d4ccf769687cd18eebd12d60e8874b/src/CharsetDetector.cs#L269

https://github.com/CharsetDetector/UTF-unknown/blob/cb3dca2a51d4ccf769687cd18eebd12d60e8874b/src/CharsetDetector.cs#L283

https://github.com/CharsetDetector/UTF-unknown/blob/cb3dca2a51d4ccf769687cd18eebd12d60e8874b/src/CharsetDetector.cs#L297-L299

**Comment by rstm-sf** [comment]
It's bug or feature?

**Comment by rstm-sf** [comment]
At least add verification will be more efficient

**Comment by rstm-sf** [comment]
It seemed like it would be better, because otherwise, you need to add a length check every time

File: tests/CharsetDetectorTest.cs [hunk 1]
```diff
@@ -60,6 +59,30 @@ public void DetectFromStreamMaxBytes(int? maxBytes, int expectedPosition, int st
             // Assert
             Assert.AreEqual(expectedPosition, stream.Position);
         }
+        
+        [Test]
+        [TestCase(new byte[] { 0x2B, 0x2F, 0x76, 0x38 })]
+        [TestCase(new byte[] { 0x2B, 0x2F, 0x76, 0x39 })]
+        [TestCase(new byte[] { 0x2B, 0x2F, 0x76, 0x2B })]
+        [TestCase(new byte[] { 0x2B, 0x2F, 0x76, 0x2F })]
+        [TestCase(new byte[] { 0x2B, 0x2F, 0x76, 0x38, 0x2D })]
```
Comment (304NotModified) [inline tests/CharsetDetectorTest.cs]: 👍 👍

**Comment by 304NotModified** [comment]
Thanks for the refactor also :)

**Pull request merged**